src/ApiService/ApiService/onefuzzlib/Storage.cs

```diff
@@ -54,40 +59,41 @@ public class Storage : IStorage {
         return _armClient;
     }
 
-    // TODO: @cached
     public IEnumerable<string> CorpusAccounts() {
-        var skip = GetFuncStorage();
-        var results = new List<string> { GetFuzzStorage() };
+        return _cache.GetOrCreate<List<string>>("CorpusAccounts", cacheEntry => {
+            var skip = GetFuncStorage();
+            var results = new List<string> { GetFuzzStorage() };
 
-        var client = GetMgmtClient();
-        var group = _creds.GetResourceGroupResourceIdentifier();
+            var client = GetMgmtClient();
+            var group = _creds.GetResourceGroupResourceIdentifier();
 
-        const string storageTypeTagKey = "storage_type";
+            const string storageTypeTagKey = "storage_type";
 
-        var resourceGroup = client.GetResourceGroupResource(group);
-        foreach (var account in resourceGroup.GetStorageAccounts()) {
-            if (account.Id == skip) {
-                continue;
-            }
+            var resourceGroup = client.GetResourceGroupResource(group);
```

Comment:
does this have an async version we can use?

Reply:
not for `GetResourceGroupResource` 
`GetStorageAccounts` does have an async version but it works at the subscription level not resource group. And there is not filtering.